### PR TITLE
Closes #37 - Fix bag with cancel alarm; Closes #70 - Add time saving when switching between tabs for the alarm

### LINKED ISSUE #37: Fix bag with cancel alarm

### PR DESCRIPTION
**Pull request by AlexLviv07**
image
image
image
image

### CODE REVIEW

**Comment by roman-fay** [comment]
style-wise my last concern is the font / circle colors. Sync with @andriy-korchynskyy and unify them

File: src/app/services/alarm.service.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  startArarm() {
+    this.alarmOn = true;
+    this.color = "blue";
+    let ms: number = this.calculateSecund();
+    let alarm = new Date(ms);
+    let differenceInMs = alarm.getTime() - (new Date().getTime());
+    this.timeOut = setTimeout(this.playAlarm.bind(this), differenceInMs);
+  }
+
+  calculateSecund() {
```
Comment (roman-fay) [inline src/app/services/alarm.service.ts]: typo: second

Comment (AlexLviv07) [reply src/app/services/alarm.service.ts]: Ok

File: src/app/alarm/alarm.component.html [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
+          <round-progress [color]="alarmService.color" [max]="60" [current]="alarmService.minute" [radius]="100"
+            [stroke]="25">
+          </round-progress>
         </div>
+      </ng-template>
+    </mat-card-content>
+    <form [formGroup]="alarmForm">
+      <div class="input-time-area">
+        <mat-form-field class="example-full-width">
+          <input matInput placeholder="Input hour" [(ngModel)]="alarmService.hour" name="Hour" formControlName="hour" [attr.disabled]="alarmService.alarmOn ?'': null">
```
Comment (roman-fay) [inline src/app/alarm/alarm.component.html]: "Set hours"

File: src/app/services/alarm.service.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AlarmService {
+  hour: string = "";
+  minute: string = "";
+  timeOut: any;
+  alarmOn: boolean = false;
+  isPlay: boolean = false;
```
Comment (roman-fay) [inline src/app/services/alarm.service.ts]: "isActive"

Comment (AlexLviv07) [reply src/app/services/alarm.service.ts]: Everything is fixed

**Pull request merged**